src/transformers/models/perceiver/modeling_perceiver.py

```diff
@@ -1830,7 +1830,7 @@ def __init__(self, config: PerceiverConfig):
             # Autoencoding, don't pass inputs to the queries.
             concat_preprocessed_input=False,
             output_shape=config.output_shape,
-            output_num_channels=512,
+            output_num_channels=config.output_num_channels,
```

Comment:
The changes here are just to enable the configuration of smaller models instead of hard-coding dims at 512 and 1024.